sql/core/src/test/resources/sql-tests/results/group-by.sql.out

```diff
@@ -708,9 +708,112 @@ FROM VALUES
  (31), (32), (33), (34), (35), (3), (37), (38), (39), (40),
  (41), (42), (43), (44), (45), (46), (47), (48), (49), (50) AS tab(col)
 -- !query schema
-struct<histogram_2:array<struct<x:double,y:double>>,histogram_3:array<struct<x:double,y:double>>,histogram_5:array<struct<x:double,y:double>>,histogram_10:array<struct<x:double,y:double>>>
+struct<histogram_2:array<struct<x:int,y:double>>,histogram_3:array<struct<x:int,y:double>>,histogram_5:array<struct<x:int,y:double>>,histogram_10:array<struct<x:int,y:double>>>
```

Comment:
We should at least update the migration guide (https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md) or even have a flag. cc @sigmod FYI

Reply:
I guess the double value behaivour is from [Hive's](https://cwiki.apache.org/confluence/display/hive/languagemanual+udf) while I like it to match the input type. I think we should have a switch to turn on and off this behaviour.

Reply:
Makes sense. I think we can follow the `Size` expression, which has a legacy config to restore to the old buggy behavior.

Reply:
Thanks! I've added the suggested flag to toggle this behavior. This seems like a reasonable config to have.